diff --git a/scripts/disnix-copy-closure.in b/scripts/disnix-copy-closure.in
index 7903434..c6b4ba1 100644
--- a/scripts/disnix-copy-closure.in
+++ b/scripts/disnix-copy-closure.in
@@ -148,7 +148,15 @@ then
 
     # Invoke a remote Disnix interface to fetch all the paths that 
     # are not already at the remote machine
-    invalidPaths=`$interface --target $target --print-invalid $requisites`
+    remaining="$(fold -sw 130000 <<< $requisites)"  # Prevent "argument list too long"
+    while [[ ${#remaining} > 0 ]]; do
+        batch="$(head -n1 <<< "$remaining")"
+        remaining="$(tail -n+2 <<< "$remaining")"
+        invalidPaths="$invalidPaths $($interface --target $target --print-invalid $batch)"
+    done
+
+    # Trim whitespace via unquoted here string
+    invalidPaths="$(cat <<< $invalidPaths)"
 
     if [ "$invalidPaths" != "" ]
     then
