diff --git a/data/lib.nix b/data/lib.nix
index e553e50..34218cf 100644
--- a/data/lib.nix
+++ b/data/lib.nix
@@ -11,7 +11,8 @@ let
   pkgs_i686_cygwin = import nixpkgs { system = "i686-cygwin"; };
   pkgs_x86_64_cygwin = import nixpkgs { system = "x86_64-cygwin"; };
   pkgs_x86_64-solaris = import nixpkgs { system = "x86_64-solaris"; };
-  
+  pkgs_armv7l-linux = import nixpkgs { system = "armv7l-linux"; };
+
 in
 rec {
   inherit (builtins) attrNames getAttr listToAttrs head tail unsafeDiscardOutputDependency hashString filter elem;
@@ -36,6 +37,7 @@ rec {
     else if system == "i686-cygwin" then pkgs_i686_cygwin
     else if system == "x86_64-cygwin" then pkgs_x86_64_cygwin
     else if system == "x86_64-solaris" then pkgs_x86_64-solaris
+    else if system == "armv7l-linux" then pkgs_armv7l-linux
     else abort "unsupported system type: ${system}";
   
   /*
