diff --git a/dysnomia-modules/process.in b/dysnomia-modules/process.in
index 455cff9..0d97467 100644
--- a/dysnomia-modules/process.in
+++ b/dysnomia-modules/process.in
@@ -137,7 +137,6 @@ case "$1" in
             if [ -e @systemd_rundir@/system/disnix-$serviceName.service ]
             then
                 removeSystemdService @systemd_rundir@/system/disnix-$serviceName.service $2
-                systemctl --system daemon-reload
             fi
         elif [ "@job_template@" = "init.d" ]
         then
diff --git a/dysnomia-modules/util b/dysnomia-modules/util
index 8b41a3f..08dd973 100644
--- a/dysnomia-modules/util
+++ b/dysnomia-modules/util
@@ -463,16 +463,19 @@ removeSystemdService()
 {
     local jobsDir=$(dirname "$1")
     local jobName=$(basename "$1" .service)
+
     
     if [ -f $2/etc/socket ]
     then
         rm -f $jobsDir/dysnomia.target.wants/$jobName.socket
         rm -f $jobsDir/$jobName.socket
     else
-        rm -f $jobsDir/dysnomia.target.wants/$jobName.service
+        systemctl disable $jobsDir/$jobName.service
     fi
     
     rm -f "$1"
+
+    systemctl --system daemon-reload
 }
 
 # Activates a systemd job and configures its wanted dependencies
@@ -511,10 +514,7 @@ activateSystemdService()
         
         startUnit=$jobName.socket
     else
-        if [ -f $jobsDir/dysnomia.target ]
-        then
-            ln -sf ../$jobName.service $jobsDir/dysnomia.target.wants/$jobName.service
-        fi
+        systemctl enable $jobsDir/$jobName.service
         
         startUnit=$jobName.service
     fi
